lcls_tools/common/data_analysis/fit/methods.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+from scipy.stats import norm, gamma
+from lcls_tools.common.data_analysis.fit.method_base import MethodBase
+
+
+class GaussianModel(MethodBase):
+    """
+    GaussianModel Class that finds initial parameter values for gaussian distribution
+    and builds probability density functions for the likelyhood a parameter
+    to be that value based on those initial parameter values.
+    Passing this class the variable profile_data automatically updates
+    the initial values and and probability density functions to match that data.
+    """
+
+    param_names: list = ["mean", "sigma", "amplitude", "offset"]
+    param_bounds: np.ndarray = np.array(
+        [
+            [0.01, 1.0],
+            [0.01, 5.0],
+            [0.01, 1.0],
+            [0.01, 1.0],
+        ]
+    )
+
+    def find_init_values(self) -> dict:
+        """Fit data without optimization, return values."""
+
+        data = self._profile_data
+        x = np.linspace(0, 1, len(data))
+        # init_fit = norm.pdf(data)
+        offset = data.min()
+        amplitude = data.max() - offset
+
+        weighted_mean = np.average(x, weights=data)
+        weighted_sigma = np.sqrt(np.cov(x, aweights=data))
+
+        self.init_values = {
+            self.param_names[0]: weighted_mean,  # data.mean()
+            self.param_names[1]: weighted_sigma,  # data.std()
+            self.param_names[2]: amplitude,
+            self.param_names[3]: offset,
+        }
+        return self.init_values
+
+    def find_priors(self) -> dict:
+        """Do initial guesses based on data and make distribution from that guess."""
+        # Creating a gamma distribution around the inital amplitude.
+        # TODO: add to comments on why gamma vs. normal dist used for priors.
+        amplitude_mean = self.init_values["amplitude"]
+        amplitude_var = 0.05
+        amplitude_alpha = (amplitude_mean**2) / amplitude_var
+        amplitude_beta = amplitude_mean / amplitude_var
+        amplitude_prior = gamma(amplitude_alpha, loc=0, scale=1 / amplitude_beta)
+
+        # Creating a normal distribution of points around the inital mean.
+        mean_prior = norm(self.init_values["mean"], 0.1)
+        # TODO: remove hard coded numbers?
+        sigma_alpha = 2.5
+        sigma_beta = 5.0
+        sigma_prior = gamma(sigma_alpha, loc=0, scale=1 / sigma_beta)
+
+        # Creating a normal distribution of points around initial offset.
+        offset_prior = norm(self.init_values["offset"], 0.5)
+        self.priors = {
+            self.param_names[0]: mean_prior,
```

Comment:
Numbered indices are not clear.

Reply:
I believe Nicole did not want hardcoded names though, so the compromise was we made a class attribute than contained the param names as a list that wouldn't get overwritten easily. I agree the using indices on that list might be confusing to a reviewer though, I am not sure what the best way to resolve both problems is.

Reply:
My understanding is that we want to avoid hard coded EPICS names. Hard coded keys of a predefined dictionary should be okay. I think defining a struct-like class could be a good solution here.